src/library/tracksuggestion/tracksuggestionfeature.cpp

```diff
@@ -0,0 +1,423 @@
+    QString propertyMessage = tr("Update the title and artist if they are misspelled.");
+
+    QStringList items;
+
+    items << tr("Import metadata from MusicBrainz")
+          << tr("Update the track properties");
+
+    QString html;
+    html.append(QString("<h2>%1</h2>").arg(title));
+    html.append(QString("<p>%1</p>").arg(status));
+    html.append(QString("<p>%1</p>").arg(updateMessage));
+    html.append(QString("<p>%1</p>").arg(propertyMessage));
+
+    return html;
+}
+
+void TrackSuggestionFeature::activate() {
+    m_title = tr("Track Suggestion");
+    emit disableSearch();
+    emit enableCoverArtDisplay(false);
+    emit switchToView("TRACKSUGGESTION");
+}
+
+void TrackSuggestionFeature::activateChild(const QModelIndex& index) {
+    TreeItem* item = static_cast<TreeItem*>(index.internalPointer());
+    if (!(item && item->getData().isValid())) {
+        return;
+    }
+    qDebug() << "TrackSuggestionFeature item" << item->getLabel()
+             << "selected. The track ID is: " << item->getData();
+    TrackId trackId = TrackId(item->getData());
+    if (!trackId.isValid()) {
+        qDebug() << "Track ID is not valid";
+        return;
+    }
+    TrackPointer trackPointer = m_pLibrary->trackCollectionManager()->getTrackById(trackId);
+    emitTrackPropertiesToDialog(trackPointer);
+    m_pTrack = trackPointer;
+    if (!lookTrackHasSuggestions(trackId)) {
+        qDebug() << "Selected track has no suggestions fetched before.";
+        emit switchToView(kSuggestionTitle);
+    } else {
+        m_pSuggestionPlaylistModel->setPlaylist(trackId.toString());
+        emit showTrackModel(m_pSuggestionPlaylistModel);
+        emit enableCoverArtDisplay(false);
+    }
+}
+
+void TrackSuggestionFeature::slotTrackChanged(const QString& group,
+        TrackPointer pNewTrack,
+        TrackPointer pOldTrack) {
+    if (pNewTrack) {
+        playerInfoTrackLoaded(group, pNewTrack);
+    }
+}
+
+void TrackSuggestionFeature::playerInfoTrackLoaded(const QString& group, TrackPointer pNewTrack) {
+    QString artist = pNewTrack->getArtist();
+    QString title = pNewTrack->getTitle();
+    TrackId trackId = pNewTrack->getId();
+    if (!trackId.isValid()) {
+        qDebug() << "Track ID is not valid!";
+        return;
+    }
+    if (group == "[Channel1]") {
+        treeItemDeckOne->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckOne->setData(trackId.toVariant());
+    } else if (group == "[Channel2]") {
+        treeItemDeckTwo->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckTwo->setData(trackId.toVariant());
+    } else if (group == "[Channel3]") {
+        treeItemDeckThree->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckThree->setData(trackId.toVariant());
+    } else {
+        treeItemDeckFour->setLabel(composeTreeItemLabel(artist, title));
+        treeItemDeckFour->setData(trackId.toVariant());
+    }
```

Comment:
I just skimmed though this and feel the redundant TreeItem code could be reduced to someting like this:
``` c++
QRegularExpression chanRegEx("^(\\[Channel)([1-4])(\\])$");
QRegularExpressionMatch chanMatch = chanRegEx.match(group);
if (!chanMatch.hasMatch()) {
    // debug output
    return;
}
TreeItem* deckTreeItem;
switch (chanMatch.captured(2).toInt()) {
case 1:
    deckTreeItem = treeItemDeckOne;
    break;
// etc.
default:
    return;
}
deckTreeItem->setLabel(composeTreeItemLabel(artist, title));
deckTreeItem->setData(trackId.toVariant());

```
It's definitely not shorter, but maybe cleaner.
Also, `[Master], num_decks` could be used instead of fixed `[1-4]`.

Note this is not a change request since you probably have more important stuff to do to finish this, but maybe it can be put on a _Polish / NiceToHave_ list for later on.

Reply:
Side note:
why are there 5 items in the last screenshot in https://github.com/mixxxdj/mixxx/pull/10860#issue-1360422117 (also used for the blogpost)?

Reply:
Hey ronso0, thanks a lot for the feedback! 🐙

QRegularExpression looks nice to have, I will definitely add in the "TO DO" list and try that. 

The new TreeItem is added just to give the user a chance to look for suggestions without placing the track on the deck. Whenever a new track is selected from library, the first TreeItem changes.

Reply:
Oh, nice feature. Is this refreshed every time another library view is selected? Because in that case the previous selection is restored and IIRC that would emit selectionChanged or something, though I'm not sure about it (and didn't test this branch, yet, obviously...)

Reply:
Thank you! That was Daniel's idea and I found it very useful so we have it :)

> Is this refreshed every time another library view is selected?

I'm sorry, I couldn't understand the question correctly. Was this question for the TreeItem connected to the chosen track? If yes, That is only changed/refreshed if a Track exists in the internal library, other than that I haven't tried it with other Libraries. As far as I remember, this wasn't related with the another libraries. Could you please correct me, if I miss-understood the question?

Reply:
```
    } else {
        treeItemDeckFour->setLabel(composeTreeItemLabel(artist, title));
        treeItemDeckFour->setData(trackId.toVariant());
    }
```
I you need to make this at least `} else if (group == "[Channel4]") {` because this also receives track load events from samplers, for example when starting Mixxx.

Reply:
> Is this refreshed every time another library view is selected?

What I meant was: is this tree item (track artist + title) updated
A) every I switch to another library view which has 1+ tracks selected?
B) only when I explicitely select a track?

I just tested, and it's A : )
Don't know if that's a bug or a feature in terms if UX.

Reply:
Yeah, I just tested it and this is discussable what would be the best for the UX.